### PR TITLE
#659: #664: fix deps

### LINKED ISSUE #659: Tomcat not working on Windows
### Expected behavior

As a IDEasy user, I want to run tomcat easily so that I can test my app in tomcat.

### Actual behavior

```
$ ide tomcat run
Configured version is 8u412b08 but does not match version to install [11,22) - need to use different version from software repository.
Using CATALINA_HOME:   D:\projects\_ide\software\default\tomcat\tomcat\10.1.30
Unable to start as CATALINA_HOME contains a colon (:) character
Step 'ide (tomcat,run)' failed: com.devonfw.tools.ide.cli.CliProcessException: Running command 'D:\projects\demo9\software\tomcat\bin\catalina.sh' using bash with arguments 'run' failed with exit code 1!
1 step(s) failed out of 2 steps.
Running command 'D:\projects\demo9\software\tomcat\bin\catalina.sh' using bash with arguments 'run' failed with exit code 1!

Error: IDEasy failed with exit code 1
```

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

1. `ide tomcat run`

### Related/Dependent Issues

* #652
* #515
* #250

### Comments/Hints:

This worked once. Was that because before we were using `catalina.bat` on Windows? I assume so.
That was my fault since I suggested for simplification that we always return `catalina.sh` instead of adding OS detection and complicated stuff. Or we did not add `CATALINA_HOME` environment variable before.

Anyhow, if we define `CATALINA_HOME` and use the bash script `catalina.sh` it seems we need to transform the PATH to MINGW syntax.

### Affected version:

* 2024.09.002-beta (will not be fixed before release we create today)

### LINKED ISSUE #664: Tool dependencies and environment variables still not working properly
### Expected behavior

As a IDEasy user, I want to install and run tools seamlessly even if they have complex dependencies so that I do not have to worry about it.

### Actual behavior

If I run the tool (e.g. tomcat) for the first time it is installed together with its dependencies.
This is also setting the environment variables and PATH extensions of dependent tools:
https://github.com/devonfw/IDEasy/blob/42201edae4aecf0687ae4c8437d0ba692401dda8/cli/src/main/java/com/devonfw/tools/ide/tool/LocalToolCommandlet.java#L372

This is called from `installTool` method but this is skipped if the tool is already installed:
https://github.com/devonfw/IDEasy/blob/42201edae4aecf0687ae4c8437d0ba692401dda8/cli/src/main/java/com/devonfw/tools/ide/tool/LocalToolCommandlet.java#L80-L84

Therefore on the first run of tomcat the environment variables will be set for tomcat including java but on the second run this does not happen.

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

1. ide tomcat
2. ide tomcat

### Related/Dependent Issues

#131 
#439

### Comments/Hints:

IMHO we have to run installTool also if the tool is already installed.
That will imply that additional things like MacOSWorkaround, etc. have to be done even if the tool is already installed wasting some performance. But fist we need to have things functionally correct and then we can think of how to optimize for performance.
After all such optimizations may save less than 1 second most probably and are therefore less important - much more important is the boostrapping performance we also waste for completion that massively slows down tab-completion that needs to be much faster whereas saving 1s on top of tool installation (check) is not very relevant.

### Affected version:

* 2024.09.002-beta-SNAPSHOT

### PR DESCRIPTION
**Pull request by hohwille**
fixes #659 
fixes #664

### CODE REVIEW

**Comment by hohwille** [comment]
Build failed with still somehow unstable or fragile IntellijTest:
```
[ERROR] Tests run: 12, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.518 s <<< FAILURE! -- in com.devonfw.tools.ide.tool.intellij.IntellijTest
[ERROR] com.devonfw.tools.ide.tool.intellij.IntellijTest.testIntellijRun(String, WireMockRuntimeInfo)[3] -- Time elapsed: 0.029 s <<< FAILURE!
java.lang.AssertionError: 

Expecting path:
  /home/runner/work/IDEasy/IDEasy/cli/target/test-projects/intellij/project/software/intellij/bin/intellijtest
to exist (symbolic links were followed).
	at com.devonfw.tools.ide.tool.intellij.IntellijTest.testIntellijRun(IntellijTest.java:136)
```
In the plain maven logs of surefire it is really hard to trace down errors that do not occur locally (on Windows).
What test value is `[3]`? Is that `linux` since it is the last value in the array?

**Review by jan-vcapgemini — APPROVED** [review]
LGTM. I've added some questions.

File: cli/src/test/java/com/devonfw/tools/ide/tool/intellij/IntellijTest.java
Comment (hohwille) [reply cli/src/test/java/com/devonfw/tools/ide/tool/intellij/IntellijTest.java:129]: I was adding this because I was desperate reading the logs on github PR build and not getting what test and OS actually caused the error.
Theoretically I could also be removed now since I was able to trace down the bug this way.
However, I am more with you to think about adding this to all these context-based tests.

Maybe we should have a look together at all of these tests and centralize a lot of such things (also asserting the same things for multiple tools, etc.). On the other hand I think that sometimes tests can get more complex to read and understand if we have too much generic stuff and over-engineer them.

File: cli/src/test/resources/ide-projects/intellij/repository/intellij/intellij/default/linux/bin/idea.sh
Comment (hohwille) [reply cli/src/test/resources/ide-projects/intellij/repository/intellij/intellij/default/linux/bin/idea.sh:4]: Oh, that `echo $PWD` was also for debugging on github.
We can discuss if we remove it here or add it to all these scripts.
I would not block this PR from merging because of this since I really want to finish the release that we post-poned multiple times already...

**Pull request merged**
